src/sage/schemes/curves/zariski_vankampen.py

```diff
@@ -927,74 +963,77 @@ def geometric_basis(G, E, EC0, p, dual_graph) -> list:
+        sage: G, E, p, EC, DG, VR = voronoi_cells(V, vertical_lines=frozenset((0 .. 4)))
+        sage: gb, vd = geometric_basis(G, E, EC, p, DG, vertical_regions=VR)
+        sage: gb
+        [[A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (-5/2, 5/2),
+          A vertex at (-1/2, 1/2), A vertex at (-1/2, -1/2), A vertex at (1/2, -1/2),
+          A vertex at (1/2, 1/2), A vertex at (-1/2, 1/2), A vertex at (-5/2, 5/2),
+          A vertex at (5/2, 5/2), A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (-5/2, 5/2),
+          A vertex at (-1/2, 1/2), A vertex at (1/2, 1/2), A vertex at (5/2, 5/2),
+          A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (5/2, 5/2), A vertex at (1/2, 1/2),
+          A vertex at (1/2, -1/2), A vertex at (5/2, -5/2)], [A vertex at (5/2, -5/2),
+          A vertex at (1/2, -1/2), A vertex at (-1/2, -1/2), A vertex at (-1/2, 1/2),
+          A vertex at (-5/2, 5/2), A vertex at (-5/2, -5/2), A vertex at (-1/2, -1/2),
+          A vertex at (1/2, -1/2), A vertex at (5/2, -5/2)],
+         [A vertex at (5/2, -5/2), A vertex at (1/2, -1/2), A vertex at (-1/2, -1/2),
+          A vertex at (-5/2, -5/2), A vertex at (5/2, -5/2)]]
+        sage: vd
+        {0: 0, 1: 3, 2: 1, 3: 2, 4: 4}
     """
     i = EC0.index(p)
-    EC = EC0[i:-1] + EC0[:i + 1]   # A counterclockwise eulerian circuit on the boundary, starting and ending at p
+    EC = EC0[i:-1] + EC0[:i + 1]
+    # A counterclockwise eulerian circuit on the boundary,
+    # starting and ending at p
     if G.size() == E.size():
         if E.is_cycle():
-            return [EC]
+            j = list(dual_graph.vertices())[0][0]
+            if j in vertical_regions:
+                vd = {j: 0}
+            else:
+                vd = {}
+            return [EC], vd
     edges_E = E.edges(sort=True)
     InternalEdges = [e for e in G.edges(sort=True) if e not in edges_E]
     InternalVertices = [v for e in InternalEdges for v in e[:2]]
     Internal = G.subgraph(vertices=InternalVertices, edges=InternalEdges)
     for i, ECi in enumerate(EC):  # q and r are the points we will cut through
         if ECi in Internal:
-            EI = [v for v in E if v in Internal.connected_component_containing_vertex(ECi, sort=True) and v != ECi]
+            EI = [v for v in E if v in
+                  Internal.connected_component_containing_vertex(ECi, sort=True)
+                  and v != ECi]
             if EI:
                 q = ECi
                 connecting_path = list(EC[:i])
                 break
         if EC[-i] in Internal:
-            EI = [v for v in E if v in Internal.connected_component_containing_vertex(EC[-i], sort=True) and v != EC[-i]]
+            EI = [v for v in E if v in
+                  Internal.connected_component_containing_vertex(EC[-i],
+                                                                 sort=True)
```

Comment:
```suggestion
                  Internal.connected_component_containing_vertex(EC[-i], sort=True)
```
Readability suffers from strict adherence to 80 chars/line.